src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +84,1326 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    s32 phi_s0;
+    f32 aux;
+    s16 theta = 0;
+    s32 i;
+
+    for (i = 0; i < 6; i++) {
+        pos.x = Math_SinS(theta) * 16.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = Math_CosS(theta) * 16.0f;
+
+        velocity.x = (pos.x * 0.6f) + (12.0f * arg3->x);
+        velocity.y = (Rand_ZeroOne() * 36.0f) + 6.0f;
+        velocity.z = (pos.z * 0.6f) + (12.0f * arg3->z);
+
+        pos.x += arg2->x;
+        pos.y += arg2->y;
+        pos.z += arg2->z;
+
+        aux = Rand_ZeroOne();
+        if (aux < 0.1f) {
+            phi_s0 = 0x61;
+        } else if (aux < 0.7f) {
+            phi_s0 = 0x41;
+        } else {
+            phi_s0 = 0x21;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, arg2, -0xC8, phi_s0, 0xA, 0xA, 0,
+                             Rand_ZeroOne() * 20.0f + 20.0f, 0x14, 0x12C, (s32)(Rand_ZeroOne() * 30.0f) + 0x1E, -1,
+                             OBJECT_GEFF, &D_06000EA0);
+
+        theta += 0x2AAA;
+    }
+
+    DemoGj_PlayExplosionSfx(this, globalCtx);
+}
+
+s32 DemoGj_IsSceneInvalid(void) {
+    if (gSaveContext.sceneSetupIndex < 4) {
+        return 0; // return false; (?)
+    }
+    return 1; // return true; (?)
+}
+
+s32 DemoGj_FindGanon(DemoGj* this, GlobalContext* globalCtx) {
+    Actor* actor;
+
+    if (this->ganon == NULL) {
+        actor = globalCtx->actorCtx.actorLists[ACTORCAT_BOSS].head;
+
+        while (actor != NULL) {
+            if (actor->id == ACTOR_BOSS_GANON2) {
+                this->ganon = (BossGanon2*)actor;
+
+                // Demo_Gj_Search_Boss_Ganon %d: Discover Ganon !!!!
+                osSyncPrintf("Demo_Gj_Search_Boss_Ganon %d:ガノン発見!!!!\n", this->dyna.actor.params);
+                return true;
+            }
+            actor = actor->next;
+        }
+
+        // Demo_Gj_Search_Boss_Ganon %d: I couldn't find Ganon
+        osSyncPrintf("Demo_Gj_Search_Boss_Ganon %d:ガノン発見出来ず\n", this->dyna.actor.params);
+        return false;
+    }
+    //! @bug: Missing return value when `this->ganon` is already set.
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Gj/func_8097BD70.s")
+void DemoGj_InitCommon(DemoGj* this, GlobalContext* globalCtx2, CollisionHeader* header) {
+    GlobalContext* globalCtx = globalCtx2;
+    Actor* actor = &this->dyna.actor;
+    DynaCollisionContext* colCtx_dyna = &globalCtx->colCtx.dyna;
+    CollisionHeader* newHeader;
+
+    if (header != NULL) {
+        Actor_ProcessInitChain(actor, sInitChain);
+        DynaPolyActor_Init(&this->dyna, DPM_UNK);
+        newHeader = NULL;
+        CollisionHeader_GetVirtual(header, &newHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, colCtx_dyna, actor, newHeader);
+    }
+}
+
+s32 DemoGj_InitSetIndexes(DemoGj* this, GlobalContext* globalCtx, s32 updateIndex, s32 drawIndex,
+                          CollisionHeader* header) {
```

Comment:
its doing more than just setting indices so im not sure i agree with the name here. but we can revisit this at a more documentation oriented stage